docs/source/java/install.rst

```diff
@@ -101,6 +101,34 @@ transitive dependencies of Flight.
         </build>
     </project>
 
+For users that consume Arrow Java through JSE17/JSE18, please remember to
+add modules needed. As an example if you are testing Java Arrow dependencies
+and seeing error like `module java.base does not opens java.nio` please
+consider to add ``--add-opens=java.base/java.nio=ALL-UNNAMED``.
```

Comment:
This is a little vague. What is needed exactly?

```suggestion
When using Java 17 or later, some JDK internals must be exposed by
adding ``--add-opens=java.base/java.nio=ALL-UNNAMED``. Otherwise,
you may see errors like ``module java.base does not "opens
java.nio" to unnamed module``.
```